controllers/storagecluster/cephblockpools.go

```diff
@@ -88,10 +88,30 @@ func (o *ocsCephBlockPools) reconcileCephBlockPool(r *StorageClusterReconciler,
 	}
 
 	_, err = ctrl.CreateOrUpdate(r.ctx, r.Client, cephBlockPool, func() error {
-		cephBlockPool.Spec.PoolSpec.DeviceClass = storageCluster.Status.DefaultCephDeviceClass
-		cephBlockPool.Spec.PoolSpec.EnableCrushUpdates = true
-		cephBlockPool.Spec.PoolSpec.FailureDomain = getFailureDomain(storageCluster)
-		cephBlockPool.Spec.PoolSpec.Replicated = generateCephReplicatedSpec(storageCluster, "data")
+		// Pass the poolSpec from the storageCluster CR
+		// Set default values in the poolSpec if necessary
+		cephBlockPool.Spec.PoolSpec = storageCluster.Spec.ManagedResources.CephBlockPools.PoolSpec
+
+		poolSpec := &cephBlockPool.Spec.PoolSpec
+		if poolSpec.DeviceClass == "" {
+			poolSpec.DeviceClass = storageCluster.Status.DefaultCephDeviceClass
+		}
+		poolSpec.EnableCrushUpdates = true
```

Comment:
Can we move fields we are not checking before setting the value before all if statements?